orderbook/src/database/fees.rs

```diff
@@ -37,69 +51,58 @@ impl MinFeeStoring for Postgres {
         &self,
         fee_data: FeeData,
         min_expiry: DateTime<Utc>,
-    ) -> Result<Option<U256>> {
+    ) -> Result<Option<UnsubsidizedFee>> {
+        // Fetches the lowest fee estimate we may have for the user
         const QUERY: &str = "\
-            SELECT MIN(min_fee) FROM min_fee_measurements \
+            SELECT gas_amount, gas_price, sell_token_price FROM min_fee_measurements \
             WHERE
                 sell_token = $1 AND \
                 buy_token = $2 AND \
                 amount = $3 AND \
                 order_kind = $4 AND \
                 expiration_timestamp >= $5 \
+            ORDER BY gas_amount * gas_price * sell_token_price ASC \
```

Comment:
Not super familiar with `sqlx`, but is there any point in adding this:

```suggestion
            ORDER BY gas_amount * gas_price * sell_token_price ASC \
            LIMIT 1
```

If `fetch_optional` just reads the first row and then stops the query, then I don't think it makes a difference.

Reply:
Does it make a difference? It was more of a question :stuck_out_tongue:.